File: leetcode/search-insert-position.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function (nums, target) {
  let bottomIndex = 0;
  let topIndex = nums.length;

  while (bottomIndex <= topIndex) {
    let middleIndex = Math.ceil((bottomIndex + topIndex) / 2);
    currentTarget = nums[middleIndex];

    if (currentTarget == target) {
      return middleIndex;
    } else if (currentTarget < target) {
      bottomIndex = middleIndex + 1;
    } else {
      topIndex = middleIndex - 1;
    }
  }

  return bottomIndex;
};

File: leetcode/search-insert-position.rs
impl Solution {
    pub fn search_insert(nums: Vec<i32>, target: i32) -> i32 {
            let mut bottom_index = 0;
    let mut top_index = nums.len() as i32 - 1;

    while bottom_index <= top_index {
        let middle_index = (bottom_index + top_index + 1) / 2;
        let current_target = nums[middle_index as usize];

        if current_target == target {
            return middle_index;
        } else if current_target < target {
            bottom_index = middle_index + 1;
        } else {
            top_index = middle_index - 1;
        }
    }

    bottom_index
    }
}